presentation/src/main/java/com/sopt/presentation/calendar/CalendarRoute.kt

```diff
@@ -92,14 +104,56 @@ fun CalendarRoute(
         )
     }
 
+    if (showSheet) {
+        NoostakBottomSheet(
+            onDismissRequest = {
+                calendarViewModel.showBottomSheet(false)
+                navController.popBackStack("first", inclusive = false)
+            },
+            content = {
+                NavHost(navController, startDestination = "first") {
+                    composable("first") {
+                        ScheduleListScreen(
+                            data = calendarViewModel.mockSchedule,
+                            onItemClick = { schedule ->
+//                                navController.currentBackStackEntry?.savedStateHandle?.set(
+//                                    "schedule",
+//                                    schedule
+//                                )
+
+                                calendarViewModel.updateDetailSchedule(schedule)
+                                navController.navigate("second")
+                            },
+                            onConfirmBtnClick = { calendarViewModel.showBottomSheet(false) }
+                        )
+                    }
+                    composable("second") { backStackEntry ->
+//                        val schedule = backStackEntry.savedStateHandle.get<ScheduleDetailEntity>("schedule")
+
```

Comment:
P2: savedStateHandle은 해당 NavBackStackEntry가 살아 있는 동안에만 유지되는데 "second"를 popBackStack 하면 "second"의 backStackEntry가 삭제되면서 저장된 데이터도 사라집니다. 따라서 "first"에서 데이터를 저장했다면, "second"의 backStackEntry가 아니라 previousBackStackEntry에서 데이터를 가져와야 합니다.
```suggestion
val schedule = navController.previousBackStackEntry?.savedStateHandle?.get<ScheduleDetailEntity>("schedule")
```

Reply:
savedStateHandle.set()을 실행하면 Intent.putExtra()를 실행한 뒤 navigate()로 이동하면 값이 전달되는 것처럼 로직이 실행된다고 생각하여 second의 backStackEntry에서 값을 가져오려는 바보 같은 짓을 했네요... 공식 문서 사이트에도 사용법이 잘 나와 있었는데 하암~... 매우 머쓱 ㅎ

Reply:
해당 화면이 없어져서 데이터가 유실된다고 해도 현재 BottomSheet이기 때문에 오히려 원하는 로직으로 구현이 되는 완전 럭키비키잖아? ㅋㅋ.. 이러한 이유로 viewModel이 아닌 savedStateHandle을 사용하겠습니당!